File: src/main/java/com/github/ovargasmahisoft/jdbcplugin/ui/OptionsForm.java
package com.github.ovargasmahisoft.jdbcplugin.ui;

import com.intellij.ide.util.PackageChooserDialog;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiPackage;

import javax.swing.*;

public class OptionsForm implements IOptionsForm {
    private JPanel mainPanel;
    private JCheckBox overwriteCheckBox;

    private JCheckBox generateDAOCheckBox;
    private JTextField basePackageTextField;
    private JButton basePackageButton;

    private JPanel daoPanel;
    private JCheckBox generateRepositoryCheckBox;
    private JCheckBox generateTestCheckBox;

    private PsiPackage basePackage;

    public OptionsForm(Project project) {

        basePackageButton.addActionListener(e -> {
            basePackage = getPackagePath(project);
            basePackageTextField.setText(basePackage.getQualifiedName());
        });

    }

    public JPanel getMainPanel() {
        return mainPanel;
    }

    @Override
    public boolean getOverwriteIfExists() {
        return overwriteCheckBox.isSelected();
    }

    @Override
    public boolean getGenerateDAO() {
        return generateDAOCheckBox.isSelected();
    }

    @Override
    public PsiPackage getBasePackage() {
        return basePackage;
    }

    @Override
    public boolean getGenerateRepository() {
        return generateRepositoryCheckBox.isSelected();
    }

    @Override
    public boolean getGenerateTestClass() {
        return generateTestCheckBox.isSelected();
    }

    private PsiPackage getPackagePath(Project project) {
        PackageChooserDialog dialog = new PackageChooserDialog("Select Package", project);
        dialog.show();
        return dialog.getSelectedPackage();
    }

}

File: src/main/kotlin/com/github/ovargasmahisoft/jdbcplugin/ui/OptionsForm.kt
package com.github.ovargasmahisoft.jdbcplugin.ui;

import javax.swing.JButton
import javax.swing.JCheckBox
import javax.swing.JPanel
import javax.swing.JTextField

interface IOptionsForm {
    fun overwriteIfExists(): Boolean
    fun generateDAO(): Boolean
    fun daoPackagePath(): String?
    fun generateRepository(): Boolean
    fun repositoryPackagePath(): String?
    fun generateGet(): Boolean
    fun generateGetMany(): Boolean
    fun generateCreate(): Boolean
    fun generateCreateMany(): Boolean
    fun generateUpdate(): Boolean
    fun generateUpdateMany(): Boolean
    fun generateDelete(): Boolean
    fun generateDeleteMany(): Boolean
}

class OptionsForm : IOptionsForm {
    lateinit var mainPanel: JPanel
    lateinit var overwriteCheckBox: JCheckBox

    lateinit var generateDAOCheckBox: JCheckBox
    lateinit var daoPackageTextField: JTextField
    lateinit var daoPackageButton: JButton

    lateinit var daoPanel: JPanel
    lateinit var generateRepositoryCheckBox: JCheckBox
    lateinit var repositoryPackageTextField: JTextField
    lateinit var repositoryPackageButton: JButton

    lateinit var getCheckBox: JCheckBox
    lateinit var getManyCheckBox: JCheckBox
    lateinit var createCheckBox: JCheckBox
    lateinit var createManyCheckBox: JCheckBox
    lateinit var updateCheckBox: JCheckBox
    lateinit var updateManyCheckBox: JCheckBox
    lateinit var deleteCheckBox: JCheckBox
    lateinit var deleteManyCheckBox: JCheckBox

    fun mainPanel() = mainPanel

    override fun overwriteIfExists() = overwriteCheckBox.isSelected

    override fun generateDAO() = generateDAOCheckBox.isSelected
    override fun daoPackagePath() = daoPackageTextField.text

    override fun generateRepository() = generateRepositoryCheckBox.isSelected
    override fun repositoryPackagePath() = repositoryPackageTextField.text

    override fun generateGet() = getCheckBox.isSelected
    override fun generateGetMany() = getManyCheckBox.isSelected

    override fun generateCreate() = createCheckBox.isSelected
    override fun generateCreateMany() = createManyCheckBox.isSelected

    override fun generateUpdate() = updateCheckBox.isSelected
    override fun generateUpdateMany() = updateManyCheckBox.isSelected

    override fun generateDelete() = deleteCheckBox.isSelected
    override fun generateDeleteMany() = deleteManyCheckBox.isSelected
}
